### DISCUSSION TITLE
runperiodic errors out on 2024.3

### DISCUSSION THREAD

**Discussion by thegcat** [pretalx/pretalx#1861] [Self-hosting] [answered]
We have updated our Pretalx installation to 2024.3.0, unfortunately the `runperiodic` cron-job now throws an error. I would appreciate pointers to debug this. The cron-job creates the following error:

```
/home/pretalx_ophase/.local/lib/python3.10/site-packages/django/db/backends/utils.py:98: RuntimeWarning: Accessing the database during app initialization is discouraged. To fix this warning, avoid executing queries in AppConfig.ready() or when your app modules are imported.
 warnings.warn(self.APPS_NOT_READY_WARNING_MSG, category=RuntimeWarning)
```

Thank you for your help.

**Comment by thegcat** [comment]
(I had also searched for issues or pull requests that might be related to this but couldn't find any, not even a closed one, thus I would suspect this to be an issue with our setup rather than it being a bug.)

**Comment by rixx** [comment]
Can you provide a more detailled error? What is the exit code, what tasks are you seeing triggered (if any), etc? From what you posted, the `runperiodic` command is printing a warning to `stdout`, but that doesn't mean it's erroring out.

**Comment by thegcat** [comment]
I just tried running it as the user instead of through cron and it worked without errors 😩 We had planned on updating the OS of that service, I'll see if the error persists afterwards.

**Reply by thegcat** [reply]
We moved to Debian 12 and run the script with a systemd timer instead of cron, the warning still shows.

**Comment by thegcat** [comment]
As an additional data point, the cron-job seems to finish through:

```
/home/pretalx_ophase/.local/lib/python3.10/site-packages/django/db/backends/utils.py:98: RuntimeWarning: Accessing the database during app initialization is discouraged. To fix this warning, avoid executing queries in AppConfig.ready() or when your app modules are imported.
 warnings.warn(self.APPS_NOT_READY_WARNING_MSG, category=RuntimeWarning)
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ [1m┏━━━━━━━━━━┓  pretalx v2024.3.0[0m                                       ┃
┃ ┃  ┌─·──╮  ┃  Settings:  /home/pretalx_ophase/.pretalx.cfg                    ┃
┃ ┃  │  O │  ┃  Database:  oh14-pretalx (postgresql)                            ┃
┃ ┃  │ ┌──╯  ┃  Logging:   /usr/share/webapps/pretalxophase/data/logs           ┃
┃ ┃  └─┘     ┃  Python:    /usr/bin/python3.10                                  ┃
┃ ┗━━━┯━┯━━━━┛  Source:    /home/pretalx_ophase/.local/lib/python3.10/site-packages/pretalx   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
```

**Reply by rixx** [reply, marked as answer]
In that case, I do agree that not showing the warning would be nice, but also that it is harmless (and here, is currently outside of our control).

**Reply by rixx** [reply]
For what it’s worth, I tried to debug this today, and it proved kinda impossible – I set a breakpoint at the indicated line, but the breakpoint never triggered and it existing made the warning go away.

After I removed the breakpoint, the warning stayed gone and I could not trigger it again, even after removing local pyc files etc. If anybody tracks down the reason for this, I’m more than happy to fix it, but until then, I’ll treat it as a spurious warning that is annoying, but also of no consequence.

**Reply by saschafoerster** [reply]
I would try the same, if the error would go away, I would be happy anyway. :) 
Could you descripte how and where to set a breakpoint?

**Reply by rixx** [reply]
I think I fixed the problem in 2124170

**Reply by saschafoerster** [reply]
@rixx Can't wait for the release. ;) (as my notification inbox still get's flooded by the cron-job-info-mails.

**Reply by rixx** [reply]
@saschafoerster I’d recommend that you silence the print output in that case, as a new release is not imminent (we may swing another one this year, but I’d give it 50:50 odds at best).

**Comment by GitHubSwimmer** [comment]
Lieber Tobias,
mein Update auf v2024.3  via docker führte auch zu einem defekt bei Pretalx.
[docker-log-pretalx.txt](https://github.com/user-attachments/files/17483649/docker-log-pretalx.txt)
Schönen Gruß

**Reply by rixx** [reply]
pretalx unterstützt, wie groß im Changelog und Blogpost steht, ab v2024.3.0 kein MySQL mehr. Du versuchst, pretalx mit MySQL laufen zu lassen. Das wird nicht funktionieren.
